libcudacxx/include/cuda/__execution/output_order.h

```diff
@@ -0,0 +1,88 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __CUDA___EXECUTION_OUTPUT_ORDER_H
+#define __CUDA___EXECUTION_OUTPUT_ORDER_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/__execution/require.h>
+#include <cuda/std/__concepts/concept_macros.h>
+#include <cuda/std/__execution/env.h>
+#include <cuda/std/__type_traits/is_one_of.h>
+
+#include <cuda/std/__cccl/prologue.h>
+
+_CCCL_BEGIN_NAMESPACE_CUDA_EXECUTION
+
+namespace output_order
+{
+
+struct __get_output_order_t;
+
+enum class __output_order_t
+{
+  __sorted,
+  __unsorted
+};
+
+template <__output_order_t _Guarantee>
+struct __output_order_holder_t : __requirement
+{
+  static constexpr __output_order_t value = _Guarantee;
```

Comment:
maybe...

```suggestion
struct _CCCL_DECLSPEC_EMPTY_BASES __output_order_holder_t
  : __requirement
  , integral_constant<__output_order_t, _Guarantee>
{
```

also, there is a downside for queries like this to all inherit from `__requirement`. it means they can't share space with each other either as empty bases or as `[[no_unique_address]]` data members. there is a rule in c++ preventing two objects of the same type from living at the same address.